tests/Feature/RateLimiterTest.php

```diff
@@ -0,0 +1,102 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Tests\Feature;
+
+use App\Http\Integrations\GithubConnector;
+use Illuminate\Foundation\Testing\RefreshDatabase;
+use Illuminate\Support\Facades\Cache;
+use Saloon\Config;
+use Saloon\Enums\Method;
+use Saloon\Http\Faking\MockClient;
+use Saloon\Http\Faking\MockResponse;
+use Saloon\Http\Request;
+use Saloon\RateLimitPlugin\Exceptions\RateLimitReachedException;
+use Tests\TestCase;
+
+class RateLimiterTest extends TestCase
+{
+    use RefreshDatabase;
+
+    protected GithubConnector $githubConnector;
+    protected string $requestUrl;
+    protected Request $request;
+
+    protected function setUp(): void
+    {
+        parent::setUp();
+        Config::preventStrayRequests();
+        Cache::clear();
+
+        $this->requestUrl = "https://api.github.com/test";
+        $this->githubConnector = new GithubConnector();
+
+        $this->request = new class() extends Request {
+            protected Method $method = Method::GET;
+
+            public function resolveEndpoint(): string
+            {
+                return "/test";
+            }
+        };
+
+        MockClient::destroyGlobal();
+    }
+
+    public function testRateLimitNotReached(): void
+    {
+        $mockClient = new MockClient([
+            $this->requestUrl => MockResponse::make(body: "", status: 200),
+        ]);
+
+        $this->githubConnector->withMockClient($mockClient);
+
+        $this->expectNotToPerformAssertions();
+
+        $this->githubConnector->send($this->request);
+    }
+
+    public function testRateLimitReached(): void
+    {
+        $mockClient = new MockClient([
+            $this->requestUrl => MockResponse::make(body: "", status: 200),
+        ]);
+
+        $this->githubConnector->withMockClient($mockClient);
+
+        $this->expectException(RateLimitReachedException::class);
+
+        for ($i = 0; $i = 5000; $i++) {
+            $this->githubConnector->send($this->request);
+        }
```

Comment:
Let's set a custom rate limiter for testing, a loop with 5000 calls looks ugly for me :vomiting_face:

Reply:
@krzysztofrewak what do u think?

Reply:
Now it looks better.